### PR TITLE
Feat add split/merge substate by clone/append.

### PR DESCRIPTION
**Pull request by tiannian**

### CODE REVIEW

File: src/state/cache.rs [hunk 1]
```diff
@@ -40,6 +40,11 @@ impl SessionedCache {
         }
     }
 
+    pub fn merge(&mut self, o: &mut Self) {
+        self.cur.append(&mut o.cur);
+        self.base.append(&mut o.cur);
```
Comment (jianwel) [inline src/state/cache.rs:45]: a note of our discussion: do not need to change the base?

File: src/state/cache.rs [hunk 2]
```diff
@@ -40,6 +40,11 @@ impl SessionedCache {
         }
     }
 
+    pub fn merge(&mut self, o: &mut Self) {
```
Comment (jianwel) [inline src/state/cache.rs:43]: since it is not currently used in any place. maybe you can add a description of the effect of this function

**Comment by kevinssgh** [comment]
Looks ok, the merge can be removed for now if there isn't a use case.

**Review by ws4charlie — APPROVED** [review]
looks good

**Pull request merged**